### ISSUE TITLE
Pattern inconsistancy for sub-requrests and pattern tester

### ISSUE THREAD

**Issue by http403** [foxyproxy/firefox-extension#85]
Hello,

I'm setting up FoxyProxy to work around some ISP limitation on certain domains, some are subsequent request by the webpage. Although I tried different settings and patterns, I couldn't get the sub request routing to work. Logs revealed that the target URL don't match any pattern I configured. I tried to use pattern tester to fix the problem but it shows some inconsistency. At the first try, it will return false but reloading it return true, no matter wildcard or regex.

However, filter set for the address bar URL works beautifully. I have no idea what went wrong.

Website I'm visiting: https://forum.worldofwarships.eu/

Here's my config
```json
{
  "iyjn5u1594226540192": {
    "type": 1,
    "color": "#66cc66",
    "title": "NordVPN",
    "active": true,
    "address": "X.nordvpn.com",
    "port": 80,
    "proxyDNS": false,
    "username": <REPLACED>,
    "password": <REPLACED>,
    "whitePatterns": [
      {
        "_comment": "This is a CDN blocked by my ISP which the website I access is going to request scripts from",
        "title": "WOW CDN",
        "pattern": "frm-wows-us.wgcdn.co",
        "type": 1,
        "protocols": 1,
        "active": true
      },
      {
        "_comment": "Yes, Foxyproxy support and helpdesk subdomain are blocked by my ISP",
        "title": "Foxyproxy Subdomain",
        "pattern": "*.getfoxyproxy.org",
        "type": 1,
        "protocols": 1,
        "active": true
      }
    ],
    "blackPatterns": [
      {
        "title": "local hostnames (usually no dots in the name). Pattern exists because 'Do not use this proxy for localhost and intranet/private IP addresses' is checked.",
        "pattern": "^(?:[^:@/]+(?::[^@/]+)?@)?(?:localhost|127\\.\\d+\\.\\d+\\.\\d+)(?::\\d+)?(?:/.*)?$",
        "type": 2,
        "protocols": 1,
        "active": true
      },
      {
        "title": "local subnets (IANA reserved address space). Pattern exists because 'Do not use this proxy for localhost and intranet/private IP addresses' is checked.",
        "pattern": "^(?:[^:@/]+(?::[^@/]+)?@)?(?:192\\.168\\.\\d+\\.\\d+|10\\.\\d+\\.\\d+\\.\\d+|172\\.(?:1[6789]|2[0-9]|3[01])\\.\\d+\\.\\d+)(?::\\d+)?(?:/.*)?$",
        "type": 2,
        "protocols": 1,
        "active": true
      },
      {
        "title": "localhost - matches the local host optionally prefixed by a user:password authentication string and optionally suffixed by a port number. The entire local subnet (127.0.0.0/8) matches. Pattern exists because 'Do not use this proxy for localhost and intranet/private IP addresses' is checked.",
        "pattern": "^(?:[^:@/]+(?::[^@/]+)?@)?[\\w-]+(?::\\d+)?(?:/.*)?$",
        "type": 2,
        "protocols": 1,
        "active": true
      }
    ],
    "pacURL": "",
    "index": 9007199254740990
  },
  "logging": {
    "size": 100,
    "active": true
  },
  "mode": "patterns",
  "browserVersion": "78.0.2",
  "foxyProxyVersion": "7.4.3",
  "foxyProxyEdition": "standard"
}
```

**Comment by ericjung** [comment]
What is the URL you are trying to match?

**Comment by http403** [comment]
I am accessing https://forum.worldofwarships.eu/ and would like to match https://frm-wows-us.wgcdn.co/.

**Comment by ericjung** [comment]
I imported your settings, turned on FoxyProxy logging, and visited [https://forum.worldofwarships.eu/](https://forum.worldofwarships.eu). i do not get routed to [https://frm-wows-us.wgcdn.co/]https://frm-wows-us.wgcdn.co. Instead, am routed to https://frm-wows-eu.wgcdn.co/ (the difference is **eu** vs **us** it seems), even though the proxy server with which I'm testing is in Los Angeles, not Europe:

![log](https://user-images.githubusercontent.com/5875610/87957440-816ef500-ca6d-11ea-8250-2db13adbcaab.png)

So I'm not able to test your exact scenario, but based on the pattern iyou're using:

`frm-wows-us.wgcdn.co`

this should work fine. Can you turn on FoxyProxy logging and tell me what you see?

**Comment by http403** [comment]
Oh yes. You're right. I just take a look on the log again and realized I clicked on the link and copied it multiple times without knowing I'm visiting the EU site rather than the US site. I must be not focusing while reading the logs from FroxyProxy and the console. 

I added `frm-wows-eu.wgcdn.co` and confirmed it worked flawlessly.

I'm so sorry in wasting your time in processing my careless mistakes. I will close this issue for good.
